### PR TITLE
amp-carousel: optional aria-label for navigation buttons

### LINKED ISSUE #442: Add ability to override aria-label in carousel
### First Timers Only
We know figuring out the process for contributing to an open source project can be intimidating, so we created this issue as a way for you to learn the ropes.  (If you feel comfortable contributing to open source projects, please leave this issue for someone else.)

This particular starter issue is designed to familiarize you with the AMP Code Contribution process by adding a simple feature to one of AMP components.

### Background
`<amp-carousel>` component has two buttons for navigating to next/previous slides. It uses `aria-label` to provide alternative text for these buttons as needed for accessibility. However the values for these labels are hard coded and not configurable.

screenshot from 2017-04-28 13 28 16

### Motivation
We like to allow page authors to optionally change the value of these labels for localization or customization purposes.

### What to do
 `amp-carousel` should accept two new attributes `data-next-button-aria-label` and  `data-previous-button-aria-label`. 

If any of these attributes are provided by the developer,  `amp-carousel` should use their values instead 
of the default strings when building the next/previous buttons in the `base-carousel.js` file.

### Step by step
- [ ] Claim this issue by adding a comment below.  Please only claim this bug if you plan on starting work in the next day or so.
- [ ] If you aren't too familiar with Git/GitHub, see the [Getting Started End-to-End Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md) for [an intro to Git & GitHub,](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#intro-to-git-and-github) and [how to get a copy of the code](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#get-a-copy-of-the-amphtml-code).  You can also refer to the [Quick Start Guide](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md) for the necessary setup steps with less explanation than the End-to-End guide.
- [ ] Follow the instructions for [building AMP](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#building-amp-and-starting-a-local-server).
- [ ] [Create a Git branch](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#create-a-git-branch) for making your changes.
- [ ] [Commit your changes](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#edit-files-and-commit-them) frequently.
- [ ] [Push your changes to GitHub](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#push-your-changes-to-your-github-fork).
- [ ] [Create a Pull Request](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#send-a-pull-request-ie-request-a-code-review).  Mention `closes Issue #442` in the description.
- [ ] Assign @aghassemi as the reviewer
- [ ] [Respond to your reviewer's comments](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#respond-to-pull-request-comments) (if any).

Once approved, your changes will be merged.  **⚡⚡⚡Congrats on making your first contribution to the AMP Project!⚡⚡⚡**  You'll be able to see it [live across the web soon](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md)!

Thanks, and we hope to see more contributions from you soon.

### Questions?

If you have questions ask in this issue or on your Pull Request (mentioning @aghassemi) or see the [How to get help](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#how-to-get-help) section of the Getting Started guide.

### PR DESCRIPTION
**Pull request by alannawalton**
With these new attributes 'data-next-button-aria-label' and 'data-previous-button-aria-label' the aria-label can be set for each of these elements.

Closes #442

### CODE REVIEW

**Comment by alannawalton** [comment]
I signed it!

File: extensions/amp-carousel/0.1/base-carousel.js [hunk 1]
```diff
@@ -13,8 +13,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import {Keycodes} from '../../../src/utils/keycodes';
-import {timerFor} from '../../../src/services';
+import {
+  Keycodes,
+} from '../../../src/utils/keycodes';
```
Comment (jridgewell) [inline extensions/amp-carousel/0.1/base-carousel.js]: You'll need to rebase, this is now `KeyCodes` from `src/utils/key-codes`

File: examples/carousel.amp.html [hunk 2]
```diff
@@ -98,7 +98,7 @@
 
   </amp-carousel>
 
-  <amp-carousel width=auto height=300 controls>
+  <amp-carousel width=auto height=300 data-next-button-aria-label="Go Forward" data-previous-button-aria-label="Go Back" controls>
```
Comment (aghassemi) [inline examples/carousel.amp.html]: nit: Let's use more descriptive values here to set a good best practice precedence. Maybe 

`Go to next slide` and `Go to previous slide`

File: extensions/amp-carousel/0.1/base-carousel.js [hunk 3]
```diff
@@ -71,7 +71,13 @@ export class BaseCarousel extends AMP.BaseElement {
     this.prevButton_.setAttribute('role', 'button');
     // TODO(erwinm): Does label need i18n support in the future? or provide
     // a way to be overridden.
-    this.prevButton_.setAttribute('aria-label', 'Previous item in carousel');
+    if (this.element.hasAttribute('data-previous-button-aria-label')) {
```
Comment (aghassemi) [inline extensions/amp-carousel/0.1/base-carousel.js]: remove the `// TODO` above since it is done

Comment (alannawalton) [inline extensions/amp-carousel/0.1/base-carousel.js]: done.

**Comment by aghassemi** [comment]
@alannawalton Looks great. I think you need to add your `@google.com` address to GitHub (see https://help.github.com/articles/adding-an-email-address-to-your-github-account/) to make the CLA check pass.

**Comment by alannawalton** [comment]
I signed it

**Comment by aghassemi** [comment]
Merged!

**Pull request merged**